packages/doenetml-worker-rust/lib-doenetml-core/src/core/document_model/dependency_creation.rs

```diff
@@ -137,7 +162,14 @@ impl DocumentModel {
                         }
                     }
                 }
-                let component_idx = resolve_prop_component(&source);
+                let component_idx = match resolve_prop_component(&source) {
+                    Ok(idx) => idx,
+                    Err(_) => {
+                        // If we can't resolve the component, then we can't resolve the prop.
+                        // Avoid a hard panic here.
+                        return linked_nodes;
```

Comment:
If we allow `add_data_query` to return a truncated set of `linked_nodes`, then when we get to `_execute_data_query_with_fresh_deps()` in `prop_calculation.rs`, then it will simply return a smaller set of results in `DataQueryResult`. This mean, in `calculate()` of a `PropUpdater`, it might get fewer results than expected. Are there scenarios where this is going to cause a problem?

If the `RequiredData` had a field of the form `PropView<type>`, then this would lead to a panic when `RequiredData::try_from_data_query_results` is unwrapped. We wouldn't want a case where we may or may not get a panic depending on what the author wrote.

I see you coded defensively in `xref.rs` by making `label` be a `Vec`. Eventually, it'd be nice if you could make it an `Option` instead. Then, if we know which types of data queries could potentially lead to failure, we could document that one should make the required data be an option.

Could we make it always panic if one codes a case where the `RequiredData` isn't an Option/Vec and the data query might return nothing?

Ideally, we want to make it hard for developers to create components that work most of the time but then fail a small percentage of time.